smartcontractkit/external-adapters-js
# Update Linear Finance EA to pull data from APIs instead of CSV files

## Closes 27595

## Description

This PR updates the Linear Finance EA to pull data from APIs instead of the manually uploaded CSV files.  It fetches the data from `https://pro-api.xangle.io/v1/index/xangle-bluechip` for bluechip data and from `https://pro-api.xangle.io/v1/index/xangle-largecap ` for largecap data.  Additionally NOPs should run the EA with the settings below to ensure up time if the API goes down.

```
WARMUP_UNHEALTHY_THRESHOLD=-1
```

## Changes

- Remove code to parse csv files
- Update Linear Finance EA to fetch data from endpoints above
- Update EA to require a `API_KEY` environment variable to be set



## Steps to Test

1. Set required environment variables and run EA.  Make sure to set a source url so that the Token Allocation EA can run as well as the Linear Finance EA depends on it.  Refer to the Token Allocation's README for instructions on what to set.
2. Send both requests below one after another and verify that both are successful

XBCI
```
{
    "id": "1",
    "data": {
        "index": "xbci",
        "source": "coinmarketcap"
    }
}
```

XLCI
```
{
    "id": "1",
    "data": {
        "index": "xlci",
        "source": "coinmarketcap"
    }
}
```

## Quality Assurance
